llsd/serde_xml.py

```diff
@@ -37,78 +68,83 @@ class LLSDXMLFormatter(LLSDBaseFormatter):
     this class since the module level format_xml() is the most convenient
     interface to this functionality.
     """
-    def _elt(self, name, contents=None):
-        """
-        Serialize a single element.
 
-        If 'contents' is omitted, write <name/>.
-        If 'contents' is bytes, write <name>contents</name>.
-        If 'contents' is str, write <name>contents.encode('utf8')</name>.
-        """
-        if not contents:
-            self.stream.writelines([b"<", name, b" />"])
-        else:
-            self.stream.writelines([b"<", name, b">",
-                                    _str_to_bytes(contents),
-                                    b"</", name, b">"])
-
-    def xml_esc(self, v):
-        "Escape string or unicode object v for xml output"
-
-        # Use is_unicode() instead of is_string() because in python 2, str is
-        # bytes, not unicode, and should not be "encode()"d. Attempts to
-        # encode("utf-8") a bytes type will result in an implicit
-        # decode("ascii") that will throw a UnicodeDecodeError if the string
-        # contains non-ascii characters.
-        if is_unicode(v):
-            # we need to drop these invalid characters because they
-            # cannot be parsed (and encode() doesn't drop them for us)
-            v = v.replace(u'\uffff', u'')
-            v = v.replace(u'\ufffe', u'')
-            v = v.encode('utf-8')
-        v = remove_invalid_xml_bytes(v)
-        return v.replace(b'&',b'&amp;').replace(b'<',b'&lt;').replace(b'>',b'&gt;')
+    def __init__(self, indent_atom = None):
+        "Construct a serializer."
+        # Call the super class constructor so that we have the type map
+        super(LLSDXMLFormatter, self).__init__()
+        self._indent_atom = b''
```

Comment:
some of the 'pretty' formatting stuff is moved into the base formatter for code sharing.  As _indent is a no-op, this is mostly non-impactful.